code/modules/projectiles/updated_projectiles/gun_attachables.dm

```diff
@@ -994,6 +994,16 @@ Defined in conflicts.dm of the #defines folder.
         return
     abstract_gun.reload(user, mag)
 
+/obj/item/attachable/attached_gun/shotgun/fire_attachment(atom/target, obj/item/weapon/gun/gun, mob/living/user)
+	if(gun.active_attachable.abstract_gun.current_mag.current_rounds > 0) //If it's still got ammo and stuff.
```

Comment:
While this works, there's no need to do it the long way. You see, this proc is being called by the attachment, so `src` and `gun.active_attachable` are the same.
That means that `gun.active_attachable.abstract_gun.current_mag.current_rounds` and `abstract_gun.current_mag.current_rounds` are the same (that last one is implicitly `src.abstract_gun.current_mag.current_rounds`).